### DISCUSSION TITLE
[@types/mui-datatables] mui-datatables 4.3 adds support for React 18

### DISCUSSION THREAD

**Discussion by ailandini-accenture** [DefinitelyTyped/DefinitelyTyped#63869] [Issues with a @types package]
Hey there!
Mui-datatables [updated to 4.3](https://github.com/gregnb/mui-datatables/releases/tag/4.3.0) to support React 18 last week (January 3) and we'd love to have the typescript compatibility updated so we can update our project.

Current working assumption is that @types/mui-datatables is still using @types/react@17
```TS2786: 'MUIDataTable' cannot be used as a JSX component.  
Its element type 'ReactElement<any, any> | Component<MUIDataTableProps, any, any> | null' is not a valid JSX element. 
    Type 'Component<MUIDataTableProps, any, any>' is not assignable to type 'Element | ElementClass | null'. 
        Type 'Component<MUIDataTableProps, any, any>' is not assignable to type 'ElementClass'. 
            The types returned by 'render()' are incompatible between these types.
                Type 'import("<project_path>/node_modules/@types/mui-datatables/node_modules/@types/react/index").ReactNode' is not assignable to type 'React.ReactNode'. 
                    Type '{}' is not assignable to type 'ReactNode'.
```

It also appears to reference the deprecated [mui npm packages](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ae894426672eed49beced9705523444b9a8f0b71/types/mui-datatables/package.json#L4-L5)

[a comment by an automated account is not shown]

**Comment by favna** [comment]
An attempt was made in the past at https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61752, however that PR ended up being closed because there was a bug in MUI. By all means, make another PR with similar changes and updating to React 18. I don't know if the MUI bug has since been solved. If not, then someone needs to bug the MUI team first.

**Comment by ailandini-accenture** [comment]
Thank you for the reply!
I have made a new fork and integrated the changes from #61752 into a [new PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63885)

I attempted to give Sam Leatherdale author credit for his portions of the code, but had trouble adding a coauthor credit due to his account being private. Please let me know if there's something I more I should do in that regard!

**Reply by favna** [reply]
> Please let me know if there's something I more I should do in that regard!

Truth be told there isn't much you can do. But if you really want to go the extra mile then you can use the following credentials with: https://github.com/git-duet/git-duet
```
name: Sam Leatherdale
email: SamLeatherdale@users.noreply.github.com
```
Data taken from <https://github.com/DefinitelyTyped/DefinitelyTyped/commit/3a0014f22219e2de771cf19be96b3370d0c550bb.patch>
